test/performance/alignment/global_affine_alignment_simd_benchmark.cpp

```diff
@@ -0,0 +1,183 @@
+                                                                      seqan3::gap_open_score{-10}}} |
+                            seqan3::align_cfg::scoring{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
+                                                                                         seqan3::mismatch_score{-5}}};
+
+// Aliases to beautify the benchmark output
+using score = seqan3::detail::with_score_type;
+using end_position = seqan3::detail::with_back_coordinate_type;
+using trace = seqan3::detail::with_alignment_type;
+
+// Globally defined constants to ensure same test data.
+inline constexpr size_t sequence_length_mean = 150;
+inline constexpr size_t set_size        = 1024;
+
+// Range to test for sequence length variance
+inline constexpr size_t deviation_begin = 0;
+inline constexpr size_t deviation_end = 64;
+inline constexpr size_t deviation_step = 8;
+
+//We don't know if the system supports hyper threading so we use only halve the threads so that the
+// simd benchmark is likely to run on physical cores only.
+uint32_t get_number_of_threads()
+{
+    uint32_t thread_count = std::thread::hardware_concurrency();
+    return (thread_count == 1) ? thread_count : thread_count >> 1;
+}
+
+// ============================================================================
+//  affine; score; dna4; collection
+// ============================================================================
+
+template <typename result_t>
+void seqan3_affine_dna4_simd(benchmark::State & state)
+{
+    size_t sequence_length_variance = state.range(0);
+    auto data = seqan3::test::generate_sequence_pairs<seqan3::dna4>(sequence_length_mean,
+                                                                    set_size,
+                                                                    sequence_length_variance);
+
+    int64_t total = 0;
+    auto vectorised_config = affine_cfg |
+                             seqan3::align_cfg::result{result_t{}, seqan3::using_score_type<int16_t>} |
+                             seqan3::align_cfg::vectorise;
+    for (auto _ : state)
+    {
+        for (auto && res : seqan3::align_pairwise(data, vectorised_config))
+            total += res.score();
+    }
+
+    state.counters["cells"] = seqan3::test::pairwise_cell_updates(data, affine_cfg);
+    state.counters["CUPS"] = seqan3::test::cell_updates_per_second(state.counters["cells"]);
+    state.counters["total"] = total;
+}
+
+BENCHMARK_TEMPLATE(seqan3_affine_dna4_simd, score)
+                        ->UseRealTime()
+                        ->DenseRange(deviation_begin, deviation_end, deviation_step);
+BENCHMARK_TEMPLATE(seqan3_affine_dna4_simd, end_position)
+                        ->UseRealTime()
+                        ->DenseRange(deviation_begin, deviation_end, deviation_step);
+
+template <typename result_t>
+void seqan3_affine_dna4_simd_parallel(benchmark::State & state)
```

Comment:
This test looks identical to the one above except the config. Why don't you give the config object as "parameter" (or however this is called in google test, data you access e.g. with `range.(0)` within the test)?

Reply:
Indeed. I slim this down. A big hooray to the configuration system of seqan3.